pingora-load-balancing/src/lib.rs

```diff
@@ -226,8 +226,10 @@ impl Backends {
                     h.observe_health(errored.is_none(), check.health_threshold(errored.is_none()));
                 if flipped {
                     if let Some(e) = errored {
+                        check.health_status_change(backend, false).await;
```

Comment:
Let's `check.health_status_change(backend, errored.is_none()).await;` before the `if let` check, then we can remove the redundant call.